elasticsearch-transport/lib/elasticsearch/transport/transport/http/faraday.rb

```diff
@@ -48,7 +48,12 @@ def __build_connection(host, options={}, block=nil)
           # @return [Array]
           #
           def host_unreachable_exceptions
-            [::Faraday::Error::ConnectionFailed, ::Faraday::Error::TimeoutError]
+            [
+                ::Faraday::ConnectionFailed,
+                ::Faraday::TimeoutError,
+                ::Faraday.const_defined?(:ServerError) ? ::Faraday::ServerError : nil,
```

Comment:
I think it would be easier to understand this if it read as:
`::Faraday.const_defined?(:ServerError) && ::Faraday::ServerError`

It would also probably be better to put this in a constant or memorized instance variable so that this array isn't created everytime a request is made... What do you think?

Reply:
> `::Faraday.const_defined?(:ServerError) && ::Faraday::ServerError`

than the `compact` trick won't do since `false` (from `const_defined?`) does not get treated as `nil` on `Array#compact`

> It would also probably be better to put this in a constant or memorized instance variable so that this array isn't created everytime a request is made... What do you think?

sure but its a pattern used through the transport classes - did not want this to get noisy.
think that would need some more refactoring affecting other http impls as well, right?

Reply:
Good point about `false` and `compact`, thanks for explaining. It makes sense to leave it the way it is then.

Ideally we would refactor the other transport classes. You're right in that would be outside the scope of this PR...so we can leave this as-is for now = )